flask_sqlalchemy/__init__.py

```diff
@@ -553,19 +561,38 @@ def get_engine(self):
             echo = self._app.config['SQLALCHEMY_ECHO']
             if (uri, echo) == self._connected_for:
                 return self._engine
-            info = make_url(uri)
-            options = {'convert_unicode': True}
-            self._sa.apply_pool_defaults(self._app, options)
-            self._sa.apply_driver_hacks(self._app, info, options)
-            if echo:
-                options['echo'] = echo
-            self._engine = rv = sqlalchemy.create_engine(info, **options)
+
+            sa_url = make_url(uri)
+            options = self.get_options(sa_url, echo)
+            self._engine = rv = self._sa.create_engine(sa_url, options)
+
             if _record_queries(self._app):
                 _EngineDebuggingSignalEvents(self._engine,
                                              self._app.import_name).register()
+
             self._connected_for = (uri, echo)
+
             return rv
 
+    def get_options(self, sa_url, echo):
+        options = {}
+        if utils.sqlalchemy_version('<', '1.3'):
```

Comment:
Reading the deprecation information on this, it seems acceptable to just remove the option altogether, rather than trying to examine SQLAlchemy's version. As their docs say, the various DB-APIs have supported Unicode on 2 and 3 for a while now.

Reply:
Done locally.